fdbserver/Ratekeeper.actor.cpp

```diff
@@ -111,8 +111,8 @@ class RatekeeperImpl {
 					    .detail("Latency", now() - self->lastSSListFetchedTimestamp);
 				}
 				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
-				std::vector<std::pair<StorageServerInterface, ProcessClass>> results =
-				    wait(getServerListAndProcessClasses(&tr));
+				std::vector<std::pair<StorageServerInterface, ProcessClass>> results = wait(timeoutError(
```

Comment:
I think the `timeout` error is not handled by `tr.onError(e)` below in the catch block, and will be thrown again, killing this actor. You can try throwing this error in `getServerListAndProcessClasses` and observe the behavior. Please add a buggified code in `getServerListAndProcessClasses` to randomly throw an error to validate the code path.

Ideally, we want this loop to retry if one SS become stuck.